src/logic/tokens/store/actions/fetchSafeTokens.ts

```diff
@@ -11,12 +11,20 @@ import { makeToken } from 'src/logic/tokens/store/model/token'
 import { TOKEN_REDUCER_ID } from 'src/logic/tokens/store/reducer/tokens'
 import updateSafe from 'src/routes/safe/store/actions/updateSafe'
 import { SAFE_REDUCER_ID } from 'src/routes/safe/store/reducer/safe'
+import { Dispatch } from 'redux'
+import { backOff } from 'exponential-backoff'
 
 const humanReadableBalance = (balance, decimals) => new BigNumber(balance).times(`1e-${decimals}`).toFixed()
 const noFunc = () => {}
 const updateSafeValue = (address) => (valueToUpdate) => updateSafe({ address, ...valueToUpdate })
 
-const fetchSafeTokens = (safeAddress) => async (dispatch, getState) => {
+let isFetchingData = false
+
+const fetchSafeTokens = (safeAddress: string) => async <T extends () => unknown>(
+  dispatch: Dispatch,
+  getState: T,
+): Promise<void> => {
+  if (isFetchingData) return
```

Comment:
shouldn't you initialize `isFetchingData = true` here?

Reply:
yes! good catch thanks, it's fixed now